lib/mongoid/extensions/hash.rb

```diff
@@ -217,10 +217,12 @@ module ClassMethods
         #
         # @param [ Object ] object The object to mongoize.
         #
-        # @return [ Hash ] The object mongoized.
+        # @return [ Hash | nil ] The object mongoized or nil.
         def mongoize(object)
           return if object.nil?
-          evolve(object.dup).transform_values!(&:mongoize)
+          if object.is_a?(Hash)
+            object.dup.transform_values!(&:mongoize)
```

Comment:
Why not https://ruby-doc.org/core-3.1.2/Hash.html#method-i-transform_values without bang, omitting `dup`?

Reply:
:+1: